pkg/eval/unix/umask.go

```diff
@@ -0,0 +1,94 @@
+// +build !windows,!plan9
+
+package unix
+
+import (
+	"errors"
+	"fmt"
+	"math"
+	"strconv"
+	"syscall"
+
+	"github.com/elves/elvish/pkg/eval"
+	"github.com/elves/elvish/pkg/eval/vars"
+)
+
+//elvdoc:var umask
+//
+// The file system creation mask. When used in a string context the output is
+// its Elvish octal representation; e.g. 0o027. This makes it possible to use
+// it in any context that expects a `$number`.
+//
+// When assigning a new value a string is implicitly treated as an octal
+// number. If that fails the usual rules for interpreting
+// [numbers](./language.html#number) are used. This means you can do both
+// `unix:umask = 027` and `unix:umask = 0o27`. You can also assign to it a
+// `float64` data type that has no fractional component. The legal values are
+// in the range [0 .. 0o777].
+//
+// You can do a temporary assignment to affect a single command; e.g.
+// `umask=077 touch a_file`. After the command completes the old umask will be
+// restored. **Warning**: Since the umask applies to the entire process, not
+// individual threads, changing it temporarily in this manner is dangerous if
+// you are doing anything in parallel. Such as via the
+// [`peach`](ref/builtin.html#peach) command.
+//
+// This variable only has meaning on UNIX like operating systems and does not
+// exist on other platforms. Do not use this in elvish scripts meant to work
+// on non-UNIX compatible systems.
+
+// UmaskVariable is a variable whose value always reflects the current file
+// creation permission mask. Setting it changes the current file creation
+// permission mask for the process (not an individual thread).
+type UmaskVariable struct {
+	ev *eval.Evaler
```

Comment:
Remove?

Reply:
Removed, although I'll note that I copied this from the *pkg/eval/pwd.go* module. I now see that it's needed in that module to support the `pwd.ev.Chdir(path)` callback. Looking at the *pkg/eval/pwd.go* module I don't understand the point of this statement:

```
var _ vars.Var = PwdVariable{}
```

Removing that statement (and the associated vars import) doesn't seem to cause any harm. The program builds and all unit tests pass.